### PR TITLE
IR Codegen: Small refactoring & comment fixes

### PR DESCRIPTION
**Pull request by Marenz**

### CODE REVIEW

File: libsolidity/codegen/ir/IRGenerator.cpp [hunk 1]
```diff
@@ -710,17 +710,17 @@ string IRGenerator::initStateVariables(ContractDefinition const& _contract)
...
-				params += baseParams.at(contract);
-		return params;
-	};
+		{
+			vector<string> params;
+			for (ContractDefinition const* contract: _contract.annotation().linearizedBaseContracts)
+				if (baseParams.count(contract))
+					params += baseParams.at(contract);
+			return params;
+		};
```
Comment (hrkrshnn) [inline libsolidity/codegen/ir/IRGenerator.cpp:723]: Why this change? I think the old indentation is correct.

File: libsolidity/codegen/ir/IRGenerator.h [hunk 2]
```diff
@@ -87,10 +87,10 @@ class IRGenerator
 	/// Generates code that assigns the initial value of the respective type.
 	std::string generateInitialAssignment(VariableDeclaration const& _varDecl);
 
-	/// Generates implicit constructors for all contracts in the inheritance hierarchy of
+	/// Generates constructors for all contracts in the inheritance hierarchy of
 	/// @a _contract
-	/// If there are user defined constructors, their body will be included in implicit constructors body.
-	void generateImplicitConstructors(ContractDefinition const& _contract);
+	/// If there are user defined constructors, their body will be included in the implicit constructors body.
```
Comment (hrkrshnn) [inline libsolidity/codegen/ir/IRGenerator.h]:
```suggestion
	/// If there are user defined constructors, their body will be included in the implicit constructors' body.
```
Optional.

File: libsolidity/codegen/ir/IRGenerator.cpp (hunk 1 above)
Comment (Marenz) [reply libsolidity/codegen/ir/IRGenerator.cpp:723]: Really? I thought it should line up with the top line

Comment (cameel) [reply libsolidity/codegen/ir/IRGenerator.cpp:723]: It should but the top line is wrapped in an unusual way. It would normally be wrapped like this:
```c++
	auto listAllParams = [&](
		map<ContractDefinition const*, vector<string>> const& baseParams
	) -> vector<string>
	{
		...
	}
```

File: libsolidity/codegen/ir/IRGenerator.cpp [hunk 3]
```diff
@@ -710,17 +710,17 @@ string IRGenerator::initStateVariables(ContractDefinition const& _contract)
 }
 
 
-void IRGenerator::generateImplicitConstructors(ContractDefinition const& _contract)
+void IRGenerator::generateConstructors(ContractDefinition const& _contract)
 {
 	auto listAllParams = [&](
 		map<ContractDefinition const*, vector<string>> const& baseParams) -> vector<string>
```
Comment (chriseth) [inline libsolidity/codegen/ir/IRGenerator.cpp]:
```suggestion
	auto listAllParams =
		[&](map<ContractDefinition const*, vector<string>> const& baseParams) -> vector<string>
```

Comment (chriseth) [reply libsolidity/codegen/ir/IRGenerator.cpp]: This way it is correct, I would say.

File: libsolidity/codegen/ir/IRGenerator.h [hunk 4]
```diff
@@ -87,10 +87,10 @@ class IRGenerator
 	/// Generates code that assigns the initial value of the respective type.
 	std::string generateInitialAssignment(VariableDeclaration const& _varDecl);
 
-	/// Generates implicit constructors for all contracts in the inheritance hierarchy of
+	/// Generates constructors for all contracts in the inheritance hierarchy of
 	/// @a _contract
-	/// If there are user defined constructors, their body will be included in implicit constructors body.
-	void generateImplicitConstructors(ContractDefinition const& _contract);
+	/// If there are user defined constructors', their body will be included in the implicit constructors body.
```
Comment (chriseth) [inline libsolidity/codegen/ir/IRGenerator.h]:
```suggestion
	/// If there are user defined constructors, their body will be included in the implicit constructor's body.
```

Comment (cameel) [inline libsolidity/codegen/ir/IRGenerator.h]:
```suggestion
	/// If there are user defined constructors, their body will be included in the implicit constructor's body.
```

**Review by cameel — COMMENTED** [review]
Looks good except for the tiny wrapping issue and a stray apostrophe.

I think the change makes sense. I would not expect `generateImplicitConstructors()` to just generate all the constructors.

File: libsolidity/codegen/ir/IRGenerator.h (hunk 2 above)
Comment (cameel) [reply libsolidity/codegen/ir/IRGenerator.h]: Oh. It's plural. Then I think it should be `bodies`:
```c++
/// If there are user defined constructors, their bodies will be included in the implicit constructors' bodies.
```

File: libsolidity/codegen/ir/IRGenerator.h (hunk 4 above)
Comment (Marenz) [reply libsolidity/codegen/ir/IRGenerator.h]: I looked it up now :P http://www.sussex.ac.uk/informatics/punctuation/apostrophe/possessives
and it seems the suggestion here is correct.

Comment (Marenz) [reply libsolidity/codegen/ir/IRGenerator.h]: I looked it up now :P http://www.sussex.ac.uk/informatics/punctuation/apostrophe/possessives
and it seems the suggestion here is correct.

File: libsolidity/codegen/ir/IRGenerator.h (hunk 2 above)
Comment (Marenz) [reply libsolidity/codegen/ir/IRGenerator.h]: If it were plural it would be `bodies` though :P

**Review by chriseth — APPROVED** [review]

File: libsolidity/codegen/ir/IRGenerator.h (hunk 2 above)
Comment (cameel) [reply libsolidity/codegen/ir/IRGenerator.h]: Yes, that's what I'm suggesting above :)

**Review by cameel — APPROVED** [review]

File: libsolidity/codegen/ir/IRGenerator.h (hunk 2 above)
Comment (Marenz) [reply libsolidity/codegen/ir/IRGenerator.h]: Oh. Well. I mean, it could be either I suppose?

Comment (cameel) [reply libsolidity/codegen/ir/IRGenerator.h]: I think it's sounds a bit weird with `body`. But I also didn't even notice that when I read the sentence first time so the meaning is clear anyway :)
